src/Symfony/Component/ArgumentResolver/ValueResolver/Traits/UidValueResolverTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ArgumentResolver\ValueResolver\Traits;
+
+use Symfony\Component\ArgumentResolver\ArgumentMetadata\ArgumentMetadata;
+use Symfony\Component\ArgumentResolver\Exception\InvalidRawValueException;
+use Symfony\Component\Uid\AbstractUid;
+
+/**
+ * @author Robin Chalas <robin@baksla.sh>
+ */
+trait UidValueResolverTrait
+{
+    private function doResolve(ArgumentMetadata $argument, array $rawValues): iterable
+    {
+        $value = $rawValues[0] ?? null;
```

Comment:
Why expecting to get an array with a `0` key here ? there is no benefit passing an array (it does not even simplify the class using the trait, as it creates the array specially for that)

Reply:
that's a leftover from a previous inheritance-based implementation of this PR, I will change it for a single value 👍